### PR TITLE
Add icon reference to Lovelace documentation

### LINKED ISSUE #16860: Link to icon documentation
### Feedback

Please link to the documentation about how icons are named, where they come from, how new ones can be added to page https://www.home-assistant.io/lovelace/dashboards-and-views

### URL

https://www.home-assistant.io/lovelace/dashboards-and-views/

### Version

2021.3.1

### PR DESCRIPTION
**Pull request by frenck**
## Proposed change

Add some information on which icons one can use to the Lovelace documentation.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #16860

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Review by pvizeli — APPROVED** [review]

File: source/lovelace/badges.markdown [hunk 1]
```diff
@@ -30,7 +30,7 @@ name:
   default: Name of entity
 icon:
   required: false
-  description: Overwrites icon or entity picture.
+  description: Overwrites icon or entity picture. You can use any icon from [MaterialDesignIcons.com](http://MaterialDesignIcons.com). Prefix the icon name with `mdi:`, ie `mdi:home`.
```
Comment (ludeeus) [inline source/lovelace/badges.markdown:33]:
```suggestion
  description: Overwrites icon or entity picture. You can use any icon from [MaterialDesignIcons.com](http://materialdesignicons.com). Prefix the icon name with `mdi:`, ie `mdi:home`.
```
URL should be lowercase

File: source/lovelace/badges.markdown [hunk 2]
```diff
@@ -87,7 +87,7 @@ name:
   type: string
 icon:
   required: false
-  description: Overwrites icon or entity picture.
+  description: Overwrites icon or entity picture. You can use any icon from [MaterialDesignIcons.com](http://MaterialDesignIcons.com). Prefix the icon name with `mdi:`, ie `mdi:home`.
```
Comment (ludeeus) [inline source/lovelace/badges.markdown:90]:
```suggestion
  description: Overwrites icon or entity picture. You can use any icon from [MaterialDesignIcons.com](http://materialdesignicons.com). Prefix the icon name with `mdi:`, ie `mdi:home`.
```
URL should be lowercase

File: source/lovelace/dashboards-and-views.markdown [hunk 3]
```diff
@@ -110,7 +110,7 @@ dashboards:
       type: string
     icon:
       required: false
-      description: The icon to show in the sidebar.
+      description: The icon to show in the sidebar. You can use any icon from [MaterialDesignIcons.com](http://MaterialDesignIcons.com). Prefix the icon name with `mdi:`, ie `mdi:home`.
```
Comment (ludeeus) [inline source/lovelace/dashboards-and-views.markdown:113]:
```suggestion
      description: The icon to show in the sidebar. You can use any icon from [MaterialDesignIcons.com](http://materialdesignicons.com). Prefix the icon name with `mdi:`, ie `mdi:home`.
```
URL should be lowercase

File: source/lovelace/dashboards-and-views.markdown [hunk 4]
```diff
@@ -227,7 +227,7 @@ views:
       default: view index
     icon:
       required: false
-      description: Icon-name from Material Design Icons.
+      description: Icon-name from Material Design Icons. You can use any icon from [MaterialDesignIcons.com](http://MaterialDesignIcons.com). Prefix the icon name with `mdi:`, ie `mdi:home`.
```
Comment (ludeeus) [inline source/lovelace/dashboards-and-views.markdown:230]:
```suggestion
      description: Icon-name from Material Design Icons. You can use any icon from [MaterialDesignIcons.com](http://materialdesignicons.com). Prefix the icon name with `mdi:`, ie `mdi:home`.
```
URL should be lowercase

File: source/lovelace/header-footer.markdown [hunk 5]
```diff
@@ -69,7 +69,7 @@ entities:
       type: string
     icon:
       required: false
-      description: Override the entity icon.
+      description: Override the entity icon. You can use any icon from [MaterialDesignIcons.com](http://MaterialDesignIcons.com). Prefix the icon name with `mdi:`, ie `mdi:home`.
```
Comment (ludeeus) [inline source/lovelace/header-footer.markdown:72]:
```suggestion
      description: Override the entity icon. You can use any icon from [MaterialDesignIcons.com](http://materialdesignicons.com). Prefix the icon name with `mdi:`, ie `mdi:home`.
```
URL should be lowercase

**Review by ludeeus — APPROVED** [review]
The statement "You can use any icon from MaterialDesignIcons.com" is technically not correct, you can only use icons that are in the version of MDI we use in the version of Core you are running, so it will be icons on MaterialDesignIcons that can not be used

**Comment by frenck** [comment]
> The statement "You can use any icon from MaterialDesignIcons.com" is technically not correct, you can only use icons that are in the version of MDI we use in the version of Core you are running, so it will be icons on MaterialDesignIcons that can not be used

True, but generally... it will be fine (as we have no other way to reference do it). Unless... we build a icon picker in the frontend of course 😉 

Comments addressed in #17012

**Comment by ludeeus** [comment]
We _could_ probably get the MDI version of the package.json in the frontend repo _somehow_ and use that :D but that will require more work than what it's worth for those few cases.

**Pull request merged**